src/index.js

```diff
@@ -114,21 +149,26 @@ class CID {
     }
 
     // otherwise, assemble the CID from the parameters
+    // type checker will not accept `version`, `codec`, `multihash` as is
+    // because types don't correspond to [number, string, Buffer] so we
+    // use any below as we know `CID.validateCID` will throw if types do not
+    // match up.
+    const [$version, $codec, $multihash]/*: any */ = [version, codec, multihash]
```

Comment:
Just do double check that I understand it correctly. This assignment makes sure that after all those if-clauses above, the variables have the right type. If e.g. `version` would now be a string, the type-checker would fail?

Reply:
Type checker without this would point out that codec is `?string` and not a `string` and `multihash` is `?Buffer` not a `Buffer` and there for can’t be assigned to fields that are `string` and `Buffer` (below)

We cast to any in the assignment so that flow just trust us what the types are and basically assigns any to `$version`, `$codec` and `$multihash`

Reply:
On nice, then it does what I expected, cool.